examples/zparkio/src/main/scala/example2.scala

```diff
@@ -0,0 +1,196 @@
+  }
+
+}
+
+package Source {
+
+  import example.example2.items.{Post, User}
+  import example.example2.services.Database
+  import example.example2.services.Database.Database
+
+  object DatabaseSource {
+    def getUsers: RIO[SparkSession with Database, Dataset[User]] =
+      // No need for implicits imports from a local variable
+      // just use : import zio.spark.sql.implicits._  at the beginning of the file
+      /* for { spark <- SparkModule() users <- { import spark.implicits._ Database[User]("SELECT * FROM users") } }
+       * yield users */
+      Database[User]("SELECT * FROM users")
+
+    def getPosts: RIO[SparkSession with Database, Dataset[Post]] =
+      // No need for implicits imports from a local variable to get the encoders for Post
+      // just use : import zio.spark.sql.implicits._  at the beginning of the file
+      /* for { spark <- SparkModule() users <- { import spark.implicits._ Database[Post]("SELECT * FROM posts") } }
+       * yield users */
+      Database[Post]("SELECT * FROM users")
+  }
+
+}
+
+package Transformations {
+
+  import example.example2.Source.DatabaseSource
+  import example.example2.items.User
+  import example.example2.services.Database.Database
+
+  object UserTransformations {
+    def getAuthors: RIO[Database with SparkSession, Dataset[User]] =
+      for {
+        /* One advantage of ZIO here is the forking of the source fetch.
+         * All source can be fetch in parallel.
+         * Without ZIO, spark would just seat there while waiting for the first source to be retrieve before sending the
+         * next query to the database. */
+        usersF <- DatabaseSource.getUsers.fork
+        postsF <- DatabaseSource.getPosts.fork
+        users  <- usersF.join
+        posts  <- postsF.join
+        authorIds <- {
+          // Dataset#collect in zio-spark is an effect. We don't need to wrap it in an attempt
+          val getAuthors = posts.map(_.authorId).distinct.collect
+          getAuthors.flatMap { ids =>
+            // we don't have broadcast mapped in zio-spark as a method
+            // we can use the retrocompat SparkSession.attempt to use existing org.apache.spark methods
+            SparkSession.attempt(_.sparkContext.broadcast(ids))
+          }
+        }
+      } yield users.filter(u => authorIds.value.contains(u.userId))
+  }
+
+}
+
+//We don't have an APP, we can use is needed zio.ZIOAppDefault
+// Roadmap for zio-spark : maybe we need to make a ZioSparkAppDefault
+trait Application /*extends ZparkioApp[Arguments, APP_ENV, Unit]*/ {
+  trait APP_ENV
+
+  protected def env: ZLayer[Arguments, Throwable, FileIO with Database] = FileIO.Live ++ Database.Live
+
+  protected def sparkFactory: SparkBuilder.type = SparkBuilder
+  // protected def loggerFactory  = zio.logging
+  // TODO : Show examples with a logger (reusing loggers in Spark with an effect system wrapper)
+
+  protected def makeCli(args: List[String]): Arguments = ??? // Arguments(args)
```

Comment:
We can replace it with zio config to show that zio-spark is well integrated into the zio ecosystem

Reply:
yep, we need an example with zio-cli + zio-config and proper logging ( #288 )